src/compas/geometry/splits/splits.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from compas.utilities import pairwise
+
+from compas.datastructures import Mesh
+from compas.geometry import Polyline
+from compas.geometry import intersection_segment_plane
+from compas.geometry import intersection_mesh_plane
+from compas.geometry import subtract_vectors
+from compas.geometry import dot_vectors
+
+__all__ = [
+    'split_polyline_plane',
+    'split_mesh_plane',
+]
+
+
+def split_polyline_plane(polyline, plane):
+    """Split a polyline by a plane, returns a list of polylines, if there has been intersections
+    Returns list of the splitted polylines
+
+    Parameters
+    ----------
+    polyline : compas.datastructures.Polyline
+    plane : compas.geometry.Plane
+
+    Returns
+    -------
+    polylines : list of compas.geometry.Polyline
+    """
+    splitted_polylines_points = []
+    sublist = []
+    for i, segment in enumerate(pairwise(polyline)):
+        sublist.append(segment.start)
+        temp_intersection = intersection_segment_plane(segment, plane)
+        if temp_intersection:
+            sublist.append(temp_intersection)
+            splitted_polylines_points.append(sublist)
+            sublist = []
+            sublist.append(temp_intersection)
+        if i == len(pairwise(polyline))-1:
+            sublist.append(segment.end)
+            splitted_polylines_points.append(sublist)
+
+    return [Polyline(sublist) for sublist in splitted_polylines_points]
+
+
+def split_mesh_plane(mesh, plane, open=True):  # compas
+    """ Calculate all the intersections between edges of the mesh and cutting plane,
+    and splits every mesh edge at the intersection point, if it exists.
+    Returns a list of the resulting splitted meshes.
+
+    Parameters
+    ----------
+    mesh : compas.datastructures.Mesh
+    plane : compas.geometry.Plane
+
+    Returns
+    -------
+    splitted_meshes : list of compas.datastructures.Mesh
+    """
+
+    def mesh_from_split(self, mesh, v_keys, f_keys, intersections, open=True):  # compas
+        """ Return a mesh from the positive or negative verts, faces and the intersection verts of a splitted mesh
+        open = true for open splits, open= False for closed splits
+        """
+        vertices = {key: mesh.vertex_coordinates(key) for key in v_keys + intersections}
+        faces = [mesh.face_vertices(f_key) for f_key in f_keys]
+        final_mesh = Mesh.from_vertices_and_faces(vertices, faces)
+        if not open:
+            final_mesh.add_face(final_mesh.vertices_on_boundary(True))
+        return final_mesh
+
+    intersections = intersection_mesh_plane(mesh, plane)
+
+    if len(intersections) > 2:
+        for f_key in list(mesh.faces()):
+            split = [v_key for v_key in mesh.face_vertices(f_key) if v_key in intersections]
+            if len(split) == 2:
+                mesh.split_face(f_key, split[0], split[1])
+
+        pos_verts = []
```

Comment:
being a bit picky here, `pos_verts` should be `positive_vertices`. Same for the negative. 
Characters are cheap these days ;)